### DISCUSSION TITLE
Authentication (Sign Up) w/ OAuth providers not working

### DISCUSSION THREAD

**Discussion by lalit2005** [supabase/supabase#1226] [Questions] [answered]
I'm just starting out with Supabase and it's really really awesome 😍 

But from past two days, I am not able to successfully finish the Sign Up part for an app 😞 😞 

I tried using both `Sign up with Google` and `Sign up with GitHub`, but both are throwing a `422 error` with the message `"Signup requires a valid password"` ...

But Sign up with email works out without throwing any error

Here is the snippet that Signs Up a user with GitHub and Google

**Sign up with Google** 👇 
```js
const [email, setEmail] = useState("");
const [password, setPassword] = useState("");

const handleSubmit = async (e) => {
e.preventDefault();
const { user, session, error } = await supabase.auth.signUp(
  {
    provider: "google",
  },
  {
    redirectTo: "http://localhost:3000/authorized",
  }
);

console.log(user, session, error);
};
```

And similarly here's the snippet for **Sign Up with GitHub** 👇 
```js
const [email, setEmail] = useState("");
const [password, setPassword] = useState("");

const handleSubmit = async (e) => {
e.preventDefault();
const { user, session, error } = await supabase.auth.signUp(
  {
    provider: "github",
  },
  {
    redirectTo: "http://localhost:3000/authorized",
  }
);

console.log(user, session, error);
};
```

**But for both of these, I get the same error** 👇 

![Screenshot](https://user-images.githubusercontent.com/69138026/115562938-a0d4d000-a2d4-11eb-9986-c4a7e20ae67a.png)
---
### Any help is appreciated and thanks in advance 😊

**Comment by easy-street** [comment, marked as answer]
from the [Sign up with third-party providers](https://supabase.io/docs/reference/javascript/auth-signup#sign-up-with-third-party-providers) example:

> You can sign up with OAuth providers using the signIn() method.

so. somewhat counterintuitively, try changing `supabase.auth.signUp` to `supabase.auth.signIn`

**Reply by lalit2005** [reply]
Oh!! My bad.
I did not notice that at all 😞
Anyway thanks a lot!!

**Comment by kpwahn** [comment]
It's also worth pointing out that if you're using "magic links" you also need to use `supabase.auth.signIn`, otherwise a password is required.
